events/api/src/main/java/org/projectnessie/events/api/Content.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.events.api;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import java.util.Collections;
+import java.util.Map;
+import org.immutables.value.Value;
+
+/** An object stored in Nessie, such as a table or a view. */
+@JsonSubTypes({
+  @JsonSubTypes.Type(name = "ICEBERG_TABLE", value = ImmutableIcebergTable.class),
+  @JsonSubTypes.Type(name = "ICEBERG_VIEW", value = ImmutableIcebergView.class),
+  @JsonSubTypes.Type(name = "NAMESPACE", value = ImmutableNamespace.class),
+  @JsonSubTypes.Type(name = "DELTA_LAKE_TABLE", value = ImmutableDeltaLakeTable.class),
+  @JsonSubTypes.Type(name = "CUSTOM", value = ImmutableCustomContent.class),
+})
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+public interface Content {
+
+  ContentType getType();
+
+  /**
+   * Unique id for this object.
+   *
+   * <p>This id is unique for the entire lifetime of this Content object and persists across
+   * renames. Two content objects with the same key will have different ids.
+   */
+  String getId();
```

Comment:
I think we can enforce `UUID` here, this seems to be the case for every new content stored in Nessie with both the old and the new storage models.

Reply:
Yes, but the main model API defines it as a `String`: https://github.com/projectnessie/nessie/blob/main/api/model/src/main/java/org/projectnessie/model/Content.java#L78

If we changes it, I think we ought to change it in both places.

@snazy , WDYT?

Reply:
Whoops I went ahead and modified it, but I can revert if necessary.

Reply:
Deviating from the `model` API on the type of content ID looks risky to me. I'd prefer reverting for now.

I'm also ok with migrating to `UUID` as the java type for content IDs everywhere (in a separate PR).

Reply:
You are right, I just reverted that.